### PR TITLE
[CONFIG] Config providers / Sequelize provider.

### LINKED ISSUE #53: Librairy suggestion - Constants
It would be nice to have a library the constants that we store in the database.
It could a bit like the Config lib, where we could have a model with decorators, and on module init, it would load all the constants from the the database.

### PR DESCRIPTION
**Pull request by Jtplouffe**
The library now has the concept of a `ConfigProvider`.
There are currently 2 implemented:
- `EnvironmentConfigProvider` (default)
- `SequelizeConfigProvider`

We can now annotate config classes with `@Config()` / `@EnvironmentConfig()` / `@SequelizeConfig()`.
After annotating the class, the config will be loaded from the right backend automatically.

If we don't annotate, it will use the environment (like the current behavior).

Here is an example of a sequelize config:

```ts
@Module({
    imports: [
        ConfigModule.forRoot({ configs: [EmaliTemplatesConfig] }),
        ConfigSequelizeModule.forRoot()
    ]
})
export class MyConfigModule {}

// ...

@SequelizeConfig()
export class EmailTemplateConfig {
    @Variable("first_email_template_id")
    firstEmailTemplateId: string;
}

// ...

@Injectable()
export class MyService {
    constructor(@InjectConfig(EmailTemplateConfig) private emailTemplateConfig: EmailTemplateConfig) {}
}
```

When using a sequelize config, a table will be automatically created (if not exists). The default table name is `configs`, but it can be overridden using `ConfigSequelizeModule.forRoot({ tableName: "my_custom_table_name" })`.

I also added an option in the `Configmodule.forRoot` to skip the variable requirement validation (for CI / test).

Configs are now also passed into class transformer, so it is now possible to transform a value (`"42"` -> `42`).

Closes #53

### CODE REVIEW

**Comment by Jtplouffe** [comment]
@JulienDuf The sequelize part has been extracted into it's own package.

**Review by JulienDuf — APPROVED** [review]

**Review by samuelgirardpaquet — APPROVED** [review]

File: config/src/handlers/config.handler.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import { ConfigMetadata } from "../models/config-metadata.model";
+import "reflect-metadata";
+import { CONFIG } from "../constant";
+import { VariableMetadata } from "../models/variable-metadata.model";
+import { Type } from "@nestjs/common";
+
+export class ConfigHandler {
+    public static getVariable(target: Type, propertyKey: string): VariableMetadata {
```
Comment (gabaudette) [inline config/src/handlers/config.handler.ts:8]:
```suggestion
    public static getVariable(target: Type, propertyKey: string): VariableMetadata | null {
```

File: config/src/models/config-metadata.model.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { VariableMetadata } from "./variable-metadata.model";
+
+export interface ConfigConfig {
```
Comment (gabaudette) [inline config/src/models/config-metadata.model.ts:3]: ConfigSetting pourrait etre un meilleur nom, non ?

File: config/src/providers/config.provider.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export interface IConfigProvider {
```
Comment (gabaudette) [inline config/src/providers/config.provider.ts:1]: Oh sh*t un "I " en prefix d'un nom d'interface

**Review by gabaudette — APPROVED** [review]

**Pull request merged**